packages/ui/LogoSettingModal/components/LogoSetting.tsx

```diff
@@ -0,0 +1,216 @@
+
+import { uploadLogo } from "../../AddLogoButton/lib/uploadLogo";
+import { Button } from "../../Button";
+import { ColorPicker } from "../../ColorPicker";
+import FileInput from "../../FileInput";
+import { Input } from "../../Input";
+import { updateProductAction } from "../actions";
+
+interface ChangeEvent extends React.ChangeEvent<HTMLInputElement> {
+  target: HTMLInputElement & EventTarget;
+}
+interface Props {
+  imageUrl: string;
+  environmentId: string;
+  setOpen?: React.Dispatch<React.SetStateAction<boolean>>;
+  setImage?: React.Dispatch<React.SetStateAction<string>>;
+  product: TProduct;
+  fromLookAndFeelSetting?: boolean;
+  imageUploadFromRegularFileUpload?: boolean;
+  setLocalProduct?: React.Dispatch<React.SetStateAction<TProduct>>;
+  setImageUrl?: React.Dispatch<React.SetStateAction<string>>;
+}
+export const LogoSetting: React.FC<Props> = ({
+  imageUrl,
+  environmentId,
+  setOpen,
+  product,
+  fromLookAndFeelSetting = false,
+  setImage,
+  imageUploadFromRegularFileUpload,
+  setLocalProduct,
+  setImageUrl,
+}) => {
+  const [backgroundColor, setBackgroundColor] = useState(product?.brand?.bgColor || "#ffffff");
+  const [isLoading, setIsLoading] = useState(false);
+  const [isStandardFileUploadOpen, setIsStandardFileUploadOpen] = useState(false);
+  const [replacedLogo, setReplacedLogo] = useState<string>(imageUrl);
+  const [isEdit, setIsEdit] = useState(imageUploadFromRegularFileUpload);
+  const replaceLogoRef = useRef<HTMLInputElement>(null);
+
+  const backgroundColorChangeHandler = (color: string) => {
+    setBackgroundColor(color);
+  };
+  const onchangeImageHandler = async (e: ChangeEvent) => {
+    const file = e.target.files?.[0];
+    if (file) {
+      await handleUpload(file, environmentId);
+    }
+  };
+  const handleUpload = async (file: File, environmentId: string) => {
+    setIsLoading(true);
+    try {
+      const { url, error } = await uploadLogo(file, environmentId);
+
+      if (error) {
+        toast.error(error);
+        setIsLoading(false);
+        return;
+      }
+
+      setReplacedLogo(url);
+      setIsStandardFileUploadOpen(false);
+    } catch (err) {
+      toast.error("Logo upload failed. Please try again.");
+      setIsLoading(false);
+    }
+    setIsLoading(false);
+  };
+
+  const handleSave = async () => {
+    try {
+      let inputProduct: Partial<TProductUpdateInput> = {
+        brand: { logoUrl: replacedLogo, bgColor: backgroundColor },
+      };
+      const updatedProductData = await updateProductAction(product.id, inputProduct);
+      setLocalProduct && setLocalProduct(updatedProductData);
+      setImageUrl && setImageUrl(replacedLogo);
+      toast.success("Logo uploaded successfully.");
+    } catch (error: any) {
+      toast.error(`Error: ${error.message}`);
+    }
+  };
+  return (
+    <>
+      <div className="relative">
+        {!isEdit && fromLookAndFeelSetting && <div className="absolute z-[999999] h-full w-full"></div>}
```

Comment:
by cleaner you mean to not use the arbitary value but a tailwind class like z-50 or something?

this is used to prevent any editing after save/cancel button is pressed in look and feel settings.